airflow/contrib/operators/awsbatch_operator.py

```diff
@@ -133,16 +134,25 @@ def _wait_for_task_ended(self):
             retry = True
             retries = 0
 
-            while retries < self.max_retries and retry:
-                self.log.info('AWS Batch retry in the next %s seconds', retries)
-                response = self.client.describe_jobs(
-                    jobs=[self.jobId]
-                )
-                if response['jobs'][-1]['status'] in ['SUCCEEDED', 'FAILED']:
+            # Allow a batch job a minute to spin up.  A random interval
+            # decreases the chances of exceeding an AWS API throttle
+            # limit when there are many concurrent tasks.
+            sleep(randint(10, 60))
+
+            while True:
+                response = self.client.describe_jobs(jobs=[self.jobId])
+                status = response['jobs'][-1]['status']
+                self.log.info('AWS Batch status: %s', status)
+                if status in ['SUCCEEDED', 'FAILED']:
                     retry = False
```

Comment:
Since we are making changes here, this loop could be tided up a bit.

```python
            while retries < self.max_retries and retry:
                response = self.client.describe_jobs(jobs=[self.jobId])
                status = response['jobs'][-1]['status']
                self.log.info('AWS Batch status: %s', status)
                if status in ['SUCCEEDED', 'FAILED']:
                    break

                retries += 1
                pause = 1 + pow(retries * 0.3, 2)
                self.log.info('AWS Batch status check (%d) in the next %.2f seconds', retries, pause)
                sleep(pause)
```

It also feels _very_ odd that if we run out of retries while the task is still running that this just falls out the bottom of the loop!

Reply:
Oh the next step is _check_success_task which will fail if it's still running.

Reply:
Thanks for the loop review and suggestion, will see if some of that can be adopted in an amended commit - but note that the suggestion now removes the status update to the `retry` variable so that bit needs a tweak.

Reply:
modified the loop and log message in the amended commit